lighthouse-core/test/audits/preload-lcp-image-test.js

```diff
@@ -0,0 +1,136 @@
+          devtoolsNodePath: '1,HTML,1,BODY,3,DIV,2,IMG',
+        },
+      ],
+      ImageElements: [
+        {
+          src: imageUrl,
+          devtoolsNodePath: '1,HTML,1,BODY,3,DIV,2,IMG',
+        },
+      ],
+    };
+  };
+
+  const mockNetworkRecords = () => {
+    return [
+      {
+        requestId: '2',
+        priority: 'High',
+        isLinkPreload: false,
+        startTime: 0,
+        endTime: 0.5,
+        timing: {receiveHeadersEnd: 500},
+        url: rootNodeUrl,
+      },
+      {
+        requestId: '2:redirect',
+        resourceType: 'Document',
+        priority: 'High',
+        isLinkPreload: false,
+        startTime: 0.5,
+        endTime: 1,
+        timing: {receiveHeadersEnd: 500},
+        url: mainDocumentNodeUrl,
+      },
+      {
+        requestId: '3',
+        resourceType: 'Script',
+        priority: 'High',
+        isLinkPreload: false,
+        startTime: 1,
+        endTime: 5,
+        timing: {receiveHeadersEnd: 4000},
+        url: scriptNodeUrl,
+        initiator: {type: 'parser', url: mainDocumentNodeUrl},
+      },
+      {
+        requestId: '4',
+        resourceType: 'Image',
+        priority: 'High',
+        isLinkPreload: false,
+        startTime: 2,
+        endTime: 4.5,
+        timing: {receiveHeadersEnd: 2500},
+        url: imageUrl,
+        initiator: {type: 'script', url: scriptNodeUrl},
+      },
+    ];
+  };
+
+  it('should suggest preloading a lcp image', async () => {
+    const networkRecords = mockNetworkRecords();
+    const artifacts = mockArtifacts(networkRecords, mainDocumentNodeUrl, imageUrl);
+    const context = {settings: {}, computedCache: new Map()};
+    const results = await PreloadLCPImage.audit(artifacts, context);
+    expect(results.numericValue).toEqual(180);
+    expect(results.details.overallSavingsMs).toEqual(180);
+    expect(results.details.items[0].url).toEqual(imageUrl);
+    expect(results.details.items[0].wastedMs).toEqual(180);
+  });
+
+  it('shouldn\'t be applicable if lcp image is not found', async () => {
+    const networkRecords = mockNetworkRecords();
+    const artifacts = mockArtifacts(networkRecords, mainDocumentNodeUrl, imageUrl);
+    artifacts.ImageElements = [];
+    const context = {settings: {}, computedCache: new Map()};
+    const results = await PreloadLCPImage.audit(artifacts, context);
+    expect(results.score).toEqual(1);
+    expect(results.notApplicable).toBeTruthy();
+    expect(results.details).toBeUndefined();
+  });
+
+  it('shouldn\'t be applicable if the lcp is not an image', async () => {
+    const artifacts = {
+      traces: {[PreloadLCPImage.DEFAULT_PASS]: lcpTrace},
+      devtoolsLogs: {[PreloadLCPImage.DEFAULT_PASS]: lcpDevtoolsLog},
+      URL: {finalUrl: 'https://www.paulirish.com/'},
+      TraceElements: [
+        {
+          traceEventType: 'largest-contentful-paint',
+          devtoolsNodePath: '1,HTML,1,BODY,3,DIV,2,P',
+        },
+      ],
+      ImageElements: [],
```

Comment:
is this different from the test before it? it'd be nice to get coverage of a clause or two in the `shouldPreloadRequest` conditions instead :)

Reply:
Oof. Didn't realize I tested the same case twice in two different ways. 🤦